src/site/_includes/components/cookie/__tests__/cookie-settings-page.test.ts

```diff
@@ -0,0 +1,62 @@
+import { fireEvent, getByRole } from '@testing-library/dom'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { initCookieSettingPage } from '..'
+import * as analytics from '../analytics'
+import { COOKIE_PREFERENCE_KEY, deleteCookie, setCookie } from '../cookie'
+
+const html = `
+<div class="js-cookie-notification-banner" hidden>Success</div>
+<form  class="js-cookie-setting-form" name="cookie-settings">
+  <label for="analytics-cookies-yes">Yes</label>
+  <input type="radio" id="analytics-cookies-yes" name="analytics-cookies" value="1" />
+  <label for="analytics-cookies-no">No</label>
+  <input type="radio" id="analytics-cookies-no" name="analytics-cookies" value="0" />
+  <button type="submit">Submit</button>
+</form>
+`
+
+vi.mock('../cookie')
+
+const FormDataMock = vi.fn((form: HTMLFormElement) => ({
```

Comment:
happy-dom doesn't have FormData class so I had to mock out a basic implementation of the `get` function I believe JSDOM use https://www.npmjs.com/package/form-data